src/plots/cartesian/set_convert.js

```diff
@@ -148,8 +148,13 @@ module.exports = function setConvert(ax, fullLayout) {
         // d2l/d2c variant that that won't add categories but will also
         // allow numbers to be mapped to the linearized axis positions
         var index;
-        if(ax._categoriesMap) { index = ax._categoriesMap[v] ? ax._categoriesMap : undefined; }
-        if(index !== undefined) { return index; }
+        if(ax._categoriesMap) {
+            index = ax._categoriesMap[v] ? ax._categoriesMap : undefined;
+            if(index !== undefined) return index;
+        } else {
+            index = ax._categories.indexOf(v);
+            if(index !== -1) return index;
+        }
```

Comment:
If this block is *only* here to fix that test, and never gets hit in real code (which looks to be the case to me, since `ax._categories` and `ax._categoriesMap` always get populated simultaneously) then we should leave out the `else` block here, and just make the test more realistic by creating the corresponding `_categoriesMap` in [the mock `ax` object](https://github.com/plotly/plotly.js/blob/master/test/jasmine/tests/axes_test.js#L1854-L1862).

For the record, this made me curious "do we still need `ax._categories` at all?" But you were right to not remove it, as `getCategoryName` (`ax.c2d` etc, ie hover info) still needs the reverse mapping to be fast.